hydroflow_lang/src/graph/flat_graph_builder.rs

```diff
@@ -44,6 +45,16 @@ pub struct FlatGraphBuilder {
 
     /// Use statements.
     uses: Vec<ItemUse>,
+
+    /// In order to make import!() statements relative to the current file, we need to know where the file is that is building the flat graph.
+    macro_invocation_path: PathBuf,
+
+    /// If the flat graph is being loaded as a module, then there are some additional things that happen
+    /// 1. the varname 'input' and 'output' is reserved and used for the input and output of the module.
+    /// 2. two initial module boundary nodes are inserted into the graph before statements are processed, one input and one output.
+    is_module: bool,
+    input_node: GraphNodeId,
+    output_node: GraphNodeId,
```

Comment:
They need to be separate nodes in the graph because otherwise removing the module boundary becomes ambiguous. You could disambiguate by doing some hacks with port names but it's easier to keep them as separate nodes imo, and it makes the flat graph look way better.

I agree with Option<GraphNodeId>

Reply:
Could be `module_nodes: Option<Ends>`

Reply:
Ehh actually Ends has redundant `Options` inside so scratch that